cloud/linode/node_controller.go

```diff
@@ -57,17 +57,29 @@ func newNodeController(kubeclient kubernetes.Interface, client client.Client, in
 }
 
 func (s *nodeController) Run(stopCh <-chan struct{}) {
-	if _, err := s.informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			node, ok := obj.(*v1.Node)
-			if !ok {
-				return
-			}
-
-			klog.Infof("NodeController will handle newly created node (%s) metadata", node.Name)
-			s.queue.Add(node)
+	if _, err := s.informer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				node, ok := obj.(*v1.Node)
+				if !ok {
+					return
+				}
+
+				klog.Infof("NodeController will handle newly created node (%s) metadata", node.Name)
+				s.queue.Add(node)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				node, ok := newObj.(*v1.Node)
+				if !ok {
+					return
+				}
+
+				klog.Infof("NodeController will handle newly updated node (%s) metadata", node.Name)
+				s.queue.Add(node)
+			},
 		},
-	}); err != nil {
+		5*time.Minute,
```

Comment:
It's a bit lost here, can we maybe make that a constant up top, so that we know about these global refresh periods and have them in one place?

Reply:
Also - the five minutes exactly the same as the TTL for the node refresh, so it's a bit racey (depending on how long the actions take).

Reply:
Fixed, and reduced the informer timeout to 1 minute.

Reply:
Cool. It's gonna be a bit spammy - it'll say "not updating due to TTL" four times in a row and then updating once, but I guess it's hard to align these time frames, especially as they'll drift anyway (due to the routines taking some time and also given that these intervals are not guaranteed to be kept exactly).